From d7bce5145bb852a9cf439326bb33e5a142b6a2e1 Mon Sep 17 00:00:00 2001
From: Roman Romanchenko <rromanchenko@sipwise.com>
Date: Thu, 29 Dec 2022 10:13:00 +0100
Subject: MT#55831 sems-* constant 10% CPU usage in idle Virtual environment vagrant/vmware

sems-* constant 10% CPU usage in idle Virtual environment vagrant/vmware
DBRegagent loop sleep replaced with waitForEvent() call
(real ticket number: TT#52455)

--- a/apps/db_reg_agent/DBRegAgent.cpp
+++ b/apps/db_reg_agent/DBRegAgent.cpp
@@ -897,9 +897,8 @@ void DBRegAgent::run() {
   DBG("running DBRegAgent thread...\n");
   shutdown_finished = false;
   while (running) {
+    waitForEventTimed(500); // 500 milliseconds
     processEvents();
-
-    usleep(1000); // 1ms
   }
 
   DBG("DBRegAgent done, removing all registrations from Event Dispatcher...\n");
--- a/core/AmEventQueue.cpp
+++ b/core/AmEventQueue.cpp
@@ -100,6 +100,11 @@ void AmEventQueue::waitForEvent()
   ev_pending.wait_for();
 }
 
+void AmEventQueue::waitForEventTimed(unsigned long msec)
+{
+  ev_pending.wait_for_to(msec);
+}
+
 void AmEventQueue::processSingleEvent()
 {
   m_queue.lock();
--- a/core/AmEventQueue.h
+++ b/core/AmEventQueue.h
@@ -80,6 +80,7 @@ public:
   void postEvent(AmEvent*);
   void processEvents();
   void waitForEvent();
+  void waitForEventTimed(unsigned long msec);
   void processSingleEvent();
   bool eventPending();
 
